### PR TITLE
Add Header Object missing attributes

### PR DESCRIPTION
**Pull request by micryc**
Refs #4196

### CODE REVIEW

File: modules/swagger-core/src/main/java/io/swagger/v3/core/util/AnnotationsUtils.java [hunk 1]
```diff
@@ -33,6 +34,7 @@
 import io.swagger.v3.oas.models.media.JsonSchema;
 import io.swagger.v3.oas.models.media.MediaType;
 import io.swagger.v3.oas.models.media.Schema;
+import io.swagger.v3.oas.models.parameters.Parameter;
```
Comment (frantuma) [inline modules/swagger-core/src/main/java/io/swagger/v3/core/util/AnnotationsUtils.java:37]: not used import?

File: modules/swagger-jaxrs2/src/test/java/io/swagger/v3/jaxrs2/annotations/operations/AnnotatedOperationMethodTest.java [hunk 2]
```diff
@@ -359,6 +362,139 @@ static class GetOperationWithResponseMultipleHeaders {
         public void simpleGet() {
         }
     }
+    @Test
```
Comment (frantuma) [inline modules/swagger-jaxrs2/src/test/java/io/swagger/v3/jaxrs2/annotations/operations/AnnotatedOperationMethodTest.java:365]: I would add a new line (applies to all added methods)

**Review by frantuma — CHANGES_REQUESTED** [review]
Seems great, only missing thing on top of unused import and new lines in testwould be - as you mentioned - support for schema.implementation / components.

As a test you could use something like:

```java
    @Test
    public void testOperationWithResponseArraySchemaImplementation() {
        Reader reader = new Reader(new OpenAPI());
        OpenAPI openAPI = reader.read(GetOperationResponseHeaderWithArraySchemaImplementation.class);
        Yaml.prettyPrint(openAPI);
        // would expect
        String expectedYAML = "openapi: 3.0.1\n" +
                "paths:\n" +
                "  /path:\n" +
                "    get:\n" +
                "      summary: Simple get operation\n" +
                "      description: Defines a simple get operation with no inputs and a complex output\n" +
                "      operationId: getWithPayloadResponse\n" +
                "      responses:\n" +
                "        \"200\":\n" +
                "          description: voila!\n" +
                "          headers:\n" +
                "            Rate-Limit-Limit:\n" +
                "              description: The number of allowed requests in the current period\n" +
                "              style: simple\n" +
                "              schema:\n" +
                "                maxItems: 10\n" +
                "                minItems: 1\n" +
                "                type: array\n" +
                "                items:\n" +
                "                  $ref: '#/components/schemas/SampleResponseSchema'\n" +
                "      deprecated: true\n" +
                "components:\n" +
                "  schemas:\n" +
                "    SampleResponseSchema:\n" +
                "      type: object\n" +
                "      properties:\n" +
                "        id:\n" +
                "          type: string\n" +
                "          description: the user id";
        assertEquals(expectedYAML, Yaml.pretty(openAPI));
    }

    static class GetOperationResponseHeaderWithArraySchemaImplementation {
        @Operation(
                summary = "Simple get operation",
                description = "Defines a simple get operation with no inputs and a complex output",
                operationId = "getWithPayloadResponse",
                deprecated = true,
                responses = {
                        @ApiResponse(
                                responseCode = "200",
                                description = "voila!",
                                headers = {@Header(
                                        name = "Rate-Limit-Limit",
                                        description = "The number of allowed requests in the current period",
                                        array = @ArraySchema(maxItems = 10, minItems = 1,schema = @Schema(implementation = SampleResponseSchema.class)))})})
        @GET
        @Path("/path")
        public void simpleGet() {
        }
    }
```

this would mean passing components from [OperationParser](https://github.com/swagger-api/swagger-core/blob/fab5000141536a5a6e9e068211fde1ec1603492b/modules/swagger-jaxrs2/src/main/java/io/swagger/v3/jaxrs2/OperationParser.java#L95) to the AnnotationUtils methods pipeline, as done in other cases. We must maintain backward compatibility, therefore new methods with new `components` param must be invoked from existing ones (passing null as param)

**Comment by micryc** [comment]
Fixes #2965 also.

**Pull request merged**